src/transformations.jl

```diff
@@ -123,23 +151,75 @@ function transform!(y::AbstractVecOrMat{<:Real}, t::ZScoreTransform, x::Abstract
                 copyto!(y, x)
             end
         else
-            broadcast!(/, y, x, s)
+            broadcast!(/, y, x, s')
         end
     else
         if isempty(s)
-            broadcast!(-, y, x, m)
+            broadcast!(-, y, x, m')
         else
-            broadcast!((x,m,s)->(x-m)/s, y, x, m, s)
+            broadcast!((x,m,s)->(x-m)/s, y, x, m', s')
         end
     end
     return y
 end
 
-function reconstruct!(x::AbstractVecOrMat{<:Real}, t::ZScoreTransform, y::AbstractVecOrMat{<:Real})
+function transform!(y::AbstractVector{<:Real}, t::ZScoreTransform, x::AbstractVector{<:Real})
+    t.dim == 1 || throw(DimensionMismatch("Inconsistent dimensions."))
+    n = size(y,1)
+    size(x,1) == n || throw(DimensionMismatch("Inconsistent dimensions."))
+
+    m = t.mean
+    s = t.scale
+
+    if isempty(m)
+        if isempty(s)
+            if x !== y
+                copyto!(y, x)
+            end
+        else
+            broadcast!(/, y, x, s[1])
+        end
+    else
+        if isempty(s)
+            broadcast!(-, y, x, m[1])
+        else
+            broadcast!((x,m,s)->(x-m)/s, y, x, m[1], s[1])
+        end
+    end
+    return y
+end
+
+function reconstruct!(x::AbstractMatrix{<:Real}, t::ZScoreTransform, y::AbstractMatrix{<:Real})
     d = t.dim
-    size(x,1) == size(y,1) == d || throw(DimensionMismatch("Inconsistent dimensions."))
-    n = size(y,2)
-    size(x,2) == n || throw(DimensionMismatch("Inconsistent dimensions."))
+    size(x,2) == size(y,2) == d || throw(DimensionMismatch("Inconsistent dimensions."))
```

Comment:
I don't understand what this is supposed to do: it compares the size of a dimension with its number.

Reply:
This exists before this PR, and I don't understand this, too.

Reply:
Ah, sorry, I misread this. `d = t.dim` gives the number of rows/columns, depending on the dimension over which standardization happens. I thought you had added it to store `dims` (which is needed, as I noted above).